backend/de.metas.business/src/main/java/de/metas/order/impl/AbstractOrderDAO.java

```diff
@@ -530,13 +530,34 @@ public Stream<I_C_Order> streamOrders(@NonNull final IQueryFilter<I_C_Order> ord
 	@Override
 	public Stream<I_C_OrderLine> streamOrderLines(@NonNull final OrderLineQuery query)
 	{
+		return toSqlQuery(query).create().stream();
+	}
+
+	private IQueryBuilder<I_C_OrderLine> toSqlQuery(@NonNull final OrderLineQuery query)
+	{
+		Check.assume(!OrderLineQuery.builder().build().equals(query), "OrderLineQuery shouldn't be empty");
```

Comment:
for the next time
* make sure OrderLineQuery is completely immutable, i.e. u can use ImmutableSet, `@singlular` knows to handle it
* introduce OrderLineQuery.EMPTY = builder().build()
* introduce OrderLineQuery.isEmpty() { return this.equals(EMPTY); }

Reply:
will include in next pr